db/external_sst_file_basic_test.cc

```diff
@@ -351,6 +353,19 @@ TEST_F(ExternalSSTFileBasicTest, BasicWithFileChecksumCrc32c) {
   DestroyAndRecreateExternalSSTFilesDir();
 }
 
+TEST_F(ExternalSSTFileBasicTest, KeyValueOrdering) {
+  Options options = CurrentOptions();
+  SstFileWriter sst_file_writer(EnvOptions(), options);
+
+  std::string file1 = sst_files_dir_ + "file01.sst";
+  ASSERT_OK(sst_file_writer.Open(file1));
+
+  ASSERT_OK(sst_file_writer.Put(Key(2000), Key(2000) + "_val"));
+  ASSERT_OK(sst_file_writer.Put(Key(2001), Key(2001) + "_val"));
+  ASSERT_OK(sst_file_writer.Put(Key(2002), Key(2002) + "_val"));
+  ASSERT_OK(sst_file_writer.Finish());
+}
```

Comment:
If there is no negative test case, we should add one.

Reply:
Thanks for pointing that out. Will add one in the next commit